src/jsdoc_transformer.ts

```diff
@@ -595,12 +595,20 @@ export function jsdocTransformer(
           }
         }
 
+        const needsExplicitThis =
+            // Methods/getters/setters/ctors already have an implicit this.
+            fnDecl.kind === ts.SyntaxKind.FunctionDeclaration ||
+            // static methods also need explicit this
+            // TODO(rado): The correct emit for static methods would be
+            // {typeof C} where C is the enclosing class.
+            (fnDecl.kind === ts.SyntaxKind.MethodDeclaration &&
```

Comment:
You can actually have static getters:
https://typescript.googleplex.com/?codez=S85JLC5WcFaoViguSSzJTFZITy1RSMvP19AEChWllpQW5SkYWivUKtRycTnrASWsAQ%3D%3D

But I believe you cannot have async getters. Maybe this code would be more readable (and that's my fault!) if we started with the "is async" condition?

Reply:
Yep, I moved the comments around so it is more clear.